src/sage/matrix/strassen.pyx

```diff
@@ -757,17 +753,16 @@ class int_range:
             sage: J * I
             [(4, 2)]
         """
-        intersection = []
         all = self.to_list()
-        for i in right.to_list():
-            if i in all:
-                intersection.append(i)
+        intersection = [i for i in right.to_list() if i in all]
```

Comment:
a faster way to test `i in all` might be better. Either make `all` a set or use binary search ?